### PR TITLE
Anchor fixes

### LINKED ISSUE #26: Error in legal expression with anchor
```
(function_declaration
  (import_path
    ((identifier) @function .)))
```

@clason @amaanq could this be the reason why Topiary formatting has problems with some anchor expressions

```
named_node [0, 0] - [2, 32]
  name: identifier [0, 1] - [0, 21]
  named_node [1, 2] - [2, 31]
    name: identifier [1, 3] - [1, 14]
    grouping [2, 4] - [2, 30]
      named_node [2, 5] - [2, 27]
        name: identifier [2, 6] - [2, 16]
        capture [2, 18] - [2, 27]
          name: identifier [2, 19] - [2, 27]
      ERROR [2, 28] - [2, 29]
```

### PR DESCRIPTION
**Pull request by amaanq**
Closes #26

### CODE REVIEW

**Comment by clason** [comment]
> #define STATE_COUNT 243
> #define STATE_COUNT 313

That's what I was worried about... As we use the parser for (potentially _very_ large) tree view in Neovim, I'm hesitant whether this is worth it?

**Comment by amaanq** [comment]
313 is nothing really, like way smaller than most other small grammars already and is only a 4kb increase in compiled size for me

File: grammar.js [hunk 1]
```diff
@@ -105,8 +106,9 @@ module.exports = grammar({
       ),
       ")",
       quantifier($),
-      captures($)
-    ),
+      captures($),
+      optional("."),
+    )),
```
[the remark this answers is not in the document]
Comment (amaanq) [reply grammar.js:111]: https://tree-sitter.github.io/tree-sitter/using-parsers

```
Similarly, an anchor placed after a pattern’s last child will cause that child pattern to only match nodes that are the last named child of their parent. The below pattern matches only nodes that are the last named child within a block.

(block (_) @last-expression .)
```

[the remark this answers is not in the document]
Comment (lucario387) [reply grammar.js:111]: That anchor is for block, not for the node with capture next to it

[the remark this answers is not in the document]
Comment (lucario387) [reply grammar.js:111]: If named node were to get this treatment, I will ask for the same for lists and anon nodes, which seems wrong

**Comment by amaanq** [comment]
back to 243 anyways

**Review by clason — APPROVED** [review]

**Pull request merged**